src/XRP/default-xrp-client.ts

```diff
@@ -466,4 +425,70 @@ export default class DefaultXrpClient implements XrpClientDecorator {
     )
     return XrpTransaction.from(getTransactionResponse, this.network)
   }
+
+  /**
+   * Populates the required fields common to all transaction types.
+   * @see https://xrpl.org/transaction-common-fields.html
+   *
+   * Note: The returned Transaction object must still be assigned transaction-specific details.
+   *
+   * @param wallet The wallet that will sign and submit this transaction.
+   * @returns A promise which resolves to a Transaction protobuf with the required common fields populated.
+   */
+  private async prepareBaseTransaction(wallet: Wallet): Promise<Transaction> {
+    const classicAddress = Utils.decodeXAddress(wallet.getAddress())
+    if (!classicAddress) {
+      throw XrpError.xAddressRequired
+    }
+    const fee = await this.getMinimumFee()
```

Comment:
Are fees on the ledger always the same?

Reply:
This is definitely an important question... great catch.  I don't think they are.  Going to do some more investigating.

Reply:
Per our offline discussion - this is the correct fee for the next transaction type we're adding (`AccountSet`).  From my notes: "there are four types of transactions that have different fee requirements: multi-signed transactions (which I suppose could be any sort of transaction because signers is a common field), EscrowFinish transactions (where the fee is proportional to the size of the fulfillment condition in bytes), key reset transaction (0 fee), and an AccountDelete transaction (flat fee of 5,000,000 drops)" -- we can leave fee population in the `prepareBaseTransaction` method and then overwrite it in the select few cases that require this (if we end up implementing them)